src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fs

```diff
@@ -17,20 +19,65 @@ type AnalyzerAttribute([<Optional; DefaultParameterValue "Analyzer">] name: stri
 
 type Context =
     {
+        FileName: string
+        SourceText: ISourceText
         ParseFileResults: FSharpParseFileResults
         CheckFileResults: FSharpCheckFileResults
-        CheckProjectResults: FSharpCheckProjectResults
-        FileName: string
-        Content: string[]
         TypedTree: FSharpImplementationFileContents
-        GetAllEntities: bool -> AssemblySymbol list
-        AllSymbolUses: FSharpSymbolUse array
-        SymbolUsesOfFile: FSharpSymbolUse array
+        CheckProjectResults: Async<FSharpCheckProjectResults>
```

Comment:
After mulling over https://github.com/ionide/FSharp.Analyzers.SDK/pull/73#issuecomment-1709336286, I think it may be better to either encode these as some form of Options or do the "tiered" approach suggested by @dawedawe (like individual parse/checkfile/checkproject events). Reason being, if this is encoded as `Async`, any analyzer can ask for this every time, and FSAC won't be in control of when to get `CheckProjectResults`. In other worse, someone is going to write an analyzer that can tank performance.